maintenance/troubleshoot/commands.md

```diff
@@ -0,0 +1,499 @@
+kube-system       coredns-66bff467f8-n95vq                   1/1     Running   0          27h
+kube-system       etcd-ip-10-0-0-10                          1/1     Running   0          27h
+kube-system       kube-apiserver-ip-10-0-0-10                1/1     Running   0          27h
+```
+
+#### Verify Kubernetes API server is running
+
+```bash
+kubectl cluster-info
+```
+```
+Kubernetes master is running at https://10.0.0.10:6443
+KubeDNS is running at https://10.0.0.10:6443/api/v1/namespaces/kube-system/services/kube-dns:dns/proxy
+ubuntu@master:~$ kubectl get svc
+NAME         TYPE        CLUSTER-IP   EXTERNAL-IP   PORT(S)   AGE
+kubernetes   ClusterIP   10.49.0.1    <none>        443/TCP   2d2h
+```
+
+#### Verify Kubernetes kube-dns is working
+
+```bash
+kubectl get svc
+```
+```
+NAME         TYPE        CLUSTER-IP   EXTERNAL-IP   PORT(S)   AGE
+kubernetes   ClusterIP   10.49.0.1    <none>        443/TCP   2d2h
+```
+
+```bash
+kubectl exec -it multitool  bash
+```
+```
+bash-5.0# curl -I -k https://kubernetes
+HTTP/2 403 
+cache-control: no-cache, private
+content-type: application/json
+x-content-type-options: nosniff
+content-length: 234
+```
+
+```bash
+bash-5.0# nslookup google.com
+```
+
+```
+Server:         10.49.0.10
+Address:        10.49.0.10#53
+Non-authoritative answer:
+Name:   google.com
+Address: 172.217.14.238
+Name:   google.com
+Address: 2607:f8b0:400a:804::200e
+```
+
+#### Verify that kubelet is running on the node with the correct flags
+
+```bash
+systemctl status kubelet
+```
+If there is a problem, check the journal
+
+```bash
+journalclt -u kubelet | head
+```
+
+#### Check the status of other system pods
+
+Look especially at coredns; if they are not getting an IP, something is wrong with the CNI
+
+```bash
+kubectl get pod -n kube-system -o wide
+```
+But if other pods fail, it is likely a different issue. Perform normal Kubernetes troubleshooting. For example:
+
+```bash
+kubectl describe pod kube-scheduler-ip-10-0-1-20.eu-west-1.compute.internal -n kube-system | tail -15
+``` 
+
+### Calico components
+
+#### View cluster CNI, IPAM, and policy 
+
+```bash
+cat /etc/cni/net.d/10-calico.conflist
+```
+
+#### Verify calicoctl matches cluster
+
+The cluster version and type must match the calicoctl version.
+
+```bash
+calicoctl version
+```
+
+For syntax:
+
+```bash
+calicoctl version -help
+```
+
+#### Check tigera operator status
+
+```bash
+kubectl get tigerastatus
+```
+
+```
+NAME     AVAILABLE   PROGRESSING   DEGRADED   SINCE
+calico   True        False         False      27h
+```
+
+#### Get tigera operator details
+
+```bash
+kubectl get tigerastatus
+```
+
+```
+NAME     AVAILABLE   PROGRESSING   DEGRADED   SINCE
+calico   True        False         False      27h
+```
+
+#### Verify component availability
+
+```bash
+kubectl get pod -n tigera-operator
+```
+
+#### View calico nodes
+
+```bash
+kubectl get pod -n calico-system -o wide
```

Comment:
We should probably note that `calico-system` is for operator based, and `kube-system` is for manifest based throughout.

Reply:
Added to top of file.